rviz_rendering/src/rviz_rendering/billboard_line.hpp

```diff
@@ -64,70 +66,74 @@ class BillboardLine : public Object
-  virtual ~BillboardLine();
+  explicit BillboardLine(Ogre::SceneManager * manager, Ogre::SceneNode * parent_node = nullptr);
+  ~BillboardLine() override;
 
   void clear();
   void newLine();
-  void addPoint(const Ogre::Vector3& point);
-  void addPoint(const Ogre::Vector3& point, const Ogre::ColourValue& color);
+  void addPoint(const Ogre::Vector3 & point);
+  void addPoint(const Ogre::Vector3 & point, const Ogre::ColourValue & color);
 
-  void setLineWidth( float width );
+  void setLineWidth(float width);
 
   void setMaxPointsPerLine(uint32_t max);
   void setNumLines(uint32_t num);
 
   // overrides from Object
-  virtual void setOrientation( const Ogre::Quaternion& orientation );
-  virtual void setPosition( const Ogre::Vector3& position );
-  virtual void setScale( const Ogre::Vector3& scale );
-  virtual void setColor( float r, float g, float b, float a );
-  virtual const Ogre::Vector3& getPosition();
-  virtual const Ogre::Quaternion& getOrientation();
+  void setOrientation(const Ogre::Quaternion & orientation) override;
+  void setPosition(const Ogre::Vector3 & position) override;
+  void setScale(const Ogre::Vector3 & scale) override;
+  void setColor(float r, float g, float b, float a) override;
+  const Ogre::Vector3 & getPosition() override;
+  const Ogre::Quaternion & getOrientation() override;
 
   /**
    * \brief Get the scene node associated with this object
    * @return The scene node associated with this object
    */
-  Ogre::SceneNode* getSceneNode() { return scene_node_; }
+  Ogre::SceneNode * getSceneNode() {return scene_node_;}
 
   /**
    * \brief We have no objects that we can set user data on
    */
-  void setUserData( const Ogre::Any& data ) {}
+  void setUserData(const Ogre::Any & data) override {(void) data;}
+
+  Ogre::MaterialPtr getMaterial() {return material_;}
 
-  Ogre::MaterialPtr getMaterial() { return material_; }
+  typedef std::vector<Ogre::BillboardChain *> V_ChainContainers;
+  /// exposed for testing
+  V_ChainContainers getChains() {return chain_containers_;}
 
 private:
-  void setupChains();
-  Ogre::BillboardChain* createChain();
+  void setupChainContainers();
+  Ogre::BillboardChain * createChain();
+  void changeAllElements(
+    std::function<Ogre::BillboardChain::Element(Ogre::BillboardChain::Element)> change_element);
+  void incrementChainContainerIfNecessary();
+  void setupChainsInChainContainers() const;
 
-  Ogre::SceneNode* scene_node_;
+  Ogre::SceneNode * scene_node_;
 
-  typedef std::vector<Ogre::BillboardChain*> V_Chain;
-  V_Chain chains_;
+  V_ChainContainers chain_containers_;
   Ogre::MaterialPtr material_;
 
   Ogre::ColourValue color_;
   float width_;
 
-  uint32_t current_line_;
-
-  // Ogre 1.4 doesn't have getNumChainElements()
+  // TODO(Martin-Idel-SI): Replace by using Ogre::BillboardChain::getNumberOfChains once available
+  // Current issue: https://github.com/OGRECave/ogre/issues/603
```

Comment:
Looks like this was fixed/merged (thanks for following up on it upstream by the way!), should we consider updating the commit hash we're tracking and fix this?

I suppose that should be done in a follow up pr. Can you make an issue to make sure we don't forget to do that?

Reply:
I'll wait for this issue/update to the pr before merging this, which every you decide.

Reply:
Yes, I believe we should do this in a new PR. I'd like to wait until we have the Windows build fully running because from past experience, I doubt updating Ogre will be a completely smooth process... I'll add an issue.